pkg/controller/daemon/daemon_controller.go

```diff
@@ -913,6 +913,32 @@ func (dsc *DaemonSetsController) podsShouldBeOnNode(
 	return nodesNeedingDaemonPods, podsToDelete
 }
 
+func (dsc *DaemonSetsController) update(ctx context.Context, ds *apps.DaemonSet, nodeList []*v1.Node, hash, key string, old []*apps.ControllerRevision) error {
```

Comment:
looks better now, than with the defer, but the name is a bit too generic IMO. What about `manageAndCleanupHistory` or something in that sense?

Reply:
or at least `updateDaemonSet`